docs/src/app/(docs)/react/utils/merge-props/page.mdx

```diff
@@ -0,0 +1,57 @@
+# mergeProps
+
+<Subtitle>A utility to merge multiple sets of React props.</Subtitle>
+<Meta
+  name="description"
+  content="A utility to merge multiple sets of React props, handling event handlers, className, and style props intelligently."
+/>
+
+`mergeProps` helps you combine multiple prop objects (for example, internal props + user props) into a single set of props you can spread onto an element.
+It behaves like `Object.assign` (rightmost wins) with a few special cases, so common React patterns work as expected.
+
+## How merging works
+
+- For most keys (everything except `className`, `style`, and event handlers), the value from the rightmost object wins:
+  ```ts
+  mergeProps({ id: 'a', dir: 'ltr' }, { id: 'b' }); // returns { id: 'b', dir: 'ltr' }
+  ```
+  Note that this includes refs.
+- `className` values are concatenated right-to-left (rightmost first):
+  ```ts
+  mergeProps({ className: 'a' }, { className: 'b' }); // className is 'b a'
+  ```
+- `style` objects are merged, with keys from the rightmost style overwriting earlier ones.
+- Event handlers are merged and executed right-to-left (rightmost first):
+
+  ```ts
+  mergeProps({ onClick: a }, { onClick: b }); // b runs before a
+  ```
+
+  - For React synthetic events, Base UI adds `event.preventBaseUIHandler()`. Calling it sets `event.baseUIHandlerPrevented = true`.
+    This does not call `preventDefault()` or `stopPropagation()`.
+    It only tells Base UI to skip earlier merged handlers or internal logic that checks the flag.
+  - For non-synthetic events (custom events with primitive/object values), this mechanism isn't available and all handlers always execute.
+
+## Passing a function instead of an object
+
+Each argument can be a props object or a function that receives the merged props up to that point (left to right) and returns a props object.
+This is useful when you need to compute the next props from whatever has already been merged.
+
+When you return event handlers from a function argument, `preventBaseUIHandler()` is not applied automatically to those handlers.
```

Comment:
A couple of code snippets on how to use the `baseUIHandlerPrevented` and `preventBaseUIHandler()` could be nice (with an object and with a function)